commons/src/main/java/fr/insideapp/sonarqube/apple/commons/antlr/SourceLinesVisitor.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * SonarQube Apple Plugin - Enables analysis of Swift and Objective-C projects into SonarQube.
+ * Copyright © 2022 inside|app (contact@insideapp.fr)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package fr.insideapp.sonarqube.apple.commons.antlr;
+
+
+import fr.insideapp.sonarqube.apple.commons.SourceLine;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.measures.CoreMetrics;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class SourceLinesVisitor implements ParseTreeItemVisitor {
```

Comment:
ake it abstract if inheriting (see ObjectiveCSourceLinesVisitor).

Reply:
Not possible : it is used by other visitors. But they do not herit from it... 
Because there is a builder in this class, and builder (inner class + static method) cannot be inherited.